### PR TITLE
use leonsim's fork of boto for fixing aws querysting issue

### PR DESCRIPTION
**Pull request by zubair-arbi**
@awais786 @tasawernawaz @waheedahmed 

Use [leonsim's fork for `boto`](https://github.com/leonsim/boto) for fixing [aws querysting issue `x-amz-security-token`](https://github.com/boto/boto/issues/1477) with hash `49b7a6e6a86f4192e7cc429cf23c427e34f36921`

This change also reverts the [previous changes](-e git+https://github.com/pmitros/DoneXBlock.git@857bf365f19c904d7e48364428f6b93ff153fabd#egg=done-xblock) by @jimabramson to resolve same issue, which didn't worked.

_Note: For local installs please uninstall the existing boto package first._

### CODE REVIEW

**Comment by zubair-arbi** [comment]
@jimabramson @maxrothman @rlucioni I have created this PR to resolve the AWS querystring issue. Please review.

**Comment by jimabramson** [comment]
@zubair-arbi i think reverting https://github.com/edx/credentials/commit/d5c28737c3317e256c73b5ea49bcefc95e880cf8 should a separate commit

File: requirements/base.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
-boto==2.32.1
+# Workaround for https://github.com/boto/boto/issues/1477
+git+https://github.com/leonsim/boto.git@49b7a6e6a86f4192e7cc429cf23c427e34f36921#egg=boto
```
Comment (jimabramson) [inline requirements/base.txt]: based on guidelines [here](https://github.com/edx/edx-platform/blob/master/requirements/edx/github.txt) I think appending `==49b7a6e6a86f4192e7cc429cf23c427e34f36921` as VERSION would be recommended, even if not absolutely necessary.

**Comment by zubair-arbi** [comment]
@jimabramson I have addressed you comments. Please review.

**Comment by jimabramson** [comment]
:+1:

**Comment by zubair-arbi** [comment]
@maxrothman since we have confirmed that this fix worked so can I merge it now?

**Comment by jimabramson** [comment]
@zubair-arbi I say we merge this and tag it for release.  I've pinged on the underlying boto issue to get a sense of whether or not we can get leonsim's fix pushed into the official repo.

**Comment by zubair-arbi** [comment]
@awais786 Please review.

**Comment by awais786** [comment]
:+1:

**Comment by zubair-arbi** [comment]
@awais786 @jimabramson @maxrothman I just checked again and the aws querysting token `x-amz-security-token` has started appearing again.

**Comment by zubair-arbi** [comment]
Ok. I understand now the `alton` is showing version for credentials as `credentials_version=d0c7db2` which is pointing to this PR `https://github.com/edx/credentials/pull/75`, so the current credentials version on `stage` is not pointing to our fix for S3.
I am going to merge this PR now.

**Pull request merged**